### ISSUE TITLE
Create initial visualisation of schools data

### ISSUE THREAD

**Issue by gilesdring** [open-innovations/leeds-2023#28]
Depends on #27 - Pushed from Sprint2 to sprint 3 as we didn't get clarity in time to take this forwards. 
Visualisations & page specs defined: https://docs.google.com/document/d/1ZdXZC95WARxoFyI1vMYiCjvNQhpbrD2SBKknzQMnjg8/edit (ask Michelle if you have any questions on this)

**Before closing item, please create a new ticket to develop the text for this page.**

**Comment by ODILMichelle** [comment]
Contact at L23 is on leave until Monday 13th, and therefore cannot identify a unique identifier for schools activities at this point in time.

**Comment by gilesdring** [comment]
Any news on this @ODILMichelle

**Comment by ODILMichelle** [comment]
We have the information outlined here which helps us understand the unique identifier for schools activities: https://github.com/open-innovations/leeds-2023/issues/27

However we are still blocked by not yet understanding exactly how L23 will be adding data to Airtable. It is still unknown whether, for events engaging pupils from multiple schools, there will be one entry for each school (allowing us to visualise pupils engaged by ward), or one entry for the event (which will not allow this). 

I don't know whether that blocks you entirely from proceeding with this work.

**Comment by gilesdring** [comment]
Let's see what the data looks like when extracted and make a call then

**Comment by ODILMichelle** [comment]
I think given the latest comment on https://github.com/open-innovations/leeds-2023/issues/27, we're likely to be able to proceed anyway

**Comment by gilesdring** [comment]
@gilesdring / @taz-chiles to discuss Tues 4th

**Comment by ODILMichelle** [comment]
Confirmed from L23:

- L23 will improve their historic data quality picking up the points we previously flagged to them
- They have added a teacher audience field which allows us to calculate true number of pupils. As this has just been added (last 10 mins or so) historic data hasn't been amended, but I will work with L23 on this. (raising another ticket shortly)
- They have created a view for us to use to capture all events (raising another ticket on this shortly)
- Previous 'test events' have been removed from the data today. This means the data we have on github is not the most up-to-date version.

**Comment by gilesdring** [comment]
Blocked by #101 and #102. Will need to update data in community engagement stats based on new output.

**Comment by taz-chiles** [comment]
I'll make a start at building the page and some initial visualisations over the next couple of weeks though, recognising that some of the data will likely change once the above updates have been made. 

Headline statistics: 
- Cumulative pupils engaged with 
- Number of schools engaged with
- Percentage of schools engaged with (not sure on how best to do this, will look into it) 

Line chart: 
- Engagement by week

Hexmap: 
- Percentage of schools engaged with in each ward (will have to look into how we can do this - would need a list of all schools per ward) 

Will catch up with @ODILMichelle next week to make sure we're happy with these being the starting point

**Comment by gilesdring** [comment]
I think this can be considered unblocked - we have a viable set of data - which might improve as the view improves or we are able to correct the transformation e.g. remove teacher data from the aggregate.

**Comment by ODILMichelle** [comment]
Regarding percentage of schools, we have a list of all schools within Leeds & their postcodes (and I believe their wards), thanks to L23. Will dig this out and/or ask me for it @taz-chiles before you start.

As added as a comment on [the agreed spec](https://docs.google.com/document/d/1ZdXZC95WARxoFyI1vMYiCjvNQhpbrD2SBKknzQMnjg8/edit), I think the Headline stat should be 'Engagements with Young People', rather than 'Pupils Engaged with', despite us agreeing the latter title with L23 previously.

**Comment by ODILMichelle** [comment]
@taz-chiles, List of schools has been added to [Sharepoint](https://odileeds.sharepoint.com/:x:/r/sites/OpenInnovationsPrivate/_layouts/15/Doc.aspx?sourcedoc=%7BF663CCFA-A834-47C1-8F7A-F1B9A4564615%7D&file=School%20Engagement%20Tracker.xlsx&cid=36f9dde9-a58f-44fd-a2f3-638f6eda0860)

I was incorrect - it's schools & wards, not schools, wards and postcodes.

**Comment by ODILMichelle** [comment]
@taz-chiles looking at the list of schools in more detail, it appears that there are some duplicates in the spreadsheet I shared above e.g. John Jamieson which is marked as both a SILC and Primary school. As such, using the total number of schools as 320 (total number of row items) will give us some incorrect results.

Happy to identify these challenges and take them back to L23 so that we can get a de-duplicated list? Let me know if this is useful.

**Comment by taz-chiles** [comment]
@ODILMichelle Thanks for digging out the spreadsheet and taking a look - yes I think going back to them would be useful , we wouldn't want to present the wrong percentage or have to make any assumptions about which rows are duplicates. I can still build out the page though, anticipating that some of the data will change.

**Comment by ODILMichelle** [comment]
Contacted L23 about the list of schools and will update when I hear back. 

I think this only blocks part of the ticket, however, and progress can still be made in terms of generating a hexmap of the numbers of young people engaged, and the first headline figure. Let me know, @taz-chiles if you disagree as I might have missed something!

**Comment by taz-chiles** [comment]
@ODILMichelle thanks for getting in touch with them! I have started building the page showing the 3 headlines and the engagement by week line chart. Will work on the hexmap this week 

@gilesdring to process the schools data, I've added a separate script that will need to be added into the DVC pipeline - I'd need to get pipenv or similar installed so that I can run and amend existing scripts with DVC so let's chat about this when you return?

**Comment by taz-chiles** [comment]
@ODILMichelle I've added some initial stats and visualisations to the page. I'm going to spend a bit of time cleaning up the code and checking over the figures, but feel free to let me know your initial comments on the page. 

I also saw that Chris has responded about the duplicate schools so I'll apply the comments from his email

**Comment by gilesdring** [comment]
Update schools reference data with schools table from airtable

**Comment by ODILMichelle** [comment]
I was about to close this ticket & noticed one small issue.

**Comment by gilesdring** [comment]
I'm looking at this at the moment to get the updated ref data. I'll pick this up too
